File: src/main/java/com/bdev/hogwarts_api/dao/StudentAttendanceDao.java
package com.bdev.hogwarts_api.dao;

import com.bdev.hogwarts_api.data.model.student_attendance.StudentAttendanceModel;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface StudentAttendanceDao extends JpaRepository<StudentAttendanceModel, Long> {
    List<StudentAttendanceModel> getAllByStudentId(long studentId);
}

File: src/main/java/com/bdev/hogwarts_api/dao/StudentAttendanceDao.kt
package com.bdev.hogwarts_api.dao

import com.bdev.hogwarts_api.data.model.student_attendance.StudentAttendanceModel
import com.bdev.hogwarts_api.data.model.student_attendance.StudentAttendanceModelId
import org.springframework.data.jpa.repository.JpaRepository

interface StudentAttendanceDao : JpaRepository<StudentAttendanceModel, StudentAttendanceModelId> {
    fun getAllByIdStudentLogin(studentLogin: String): List<StudentAttendanceModel>
}